src/main/java/io/odpf/firehose/sink/SinkFactory.java

```diff
@@ -68,6 +70,8 @@ public void init() {
     public Sink getSink() {
         instrumentation.logInfo("Sink Type: {}", kafkaConsumerConfig.getSinkType().toString());
         switch (kafkaConsumerConfig.getSinkType()) {
+            case CLICKHOUSE:
+                return ClickhouseSinkFactory.create(config, statsDReporter, stencilClient);
```

Comment:
Do you think Clickhouse Sink can be used with FirehoseAsyncConsumer? If yes then, can we think about a common initialisation of the ClickhouseSinkfactory?

Reply:
i meant if there are some common variables which can be used across multiple Sink objects, we can have init() function called only once.

Reply:
I have eliminated two common variables Parser and ProtoToFieldMapper.

Reply:
Done.